src/keeper.js

```diff
@@ -59,25 +59,30 @@ export default class keeper {
         this._activeAuctions = await clip.activeAuctions();
         auction = this._activeAuctions[i];
 
-
-        const lot = (auction.lot.toString());
-        // Pass in the entire auction size into Uniswap and store the Dai proceeds form the trade
-        await uniswap.fetch(lot);
-        // Find the minimum effective exchange rate between collateral/Dai
-        // e.x. ETH price 1000 DAI -> minimum profit of 1% -> new ETH price is 1000*1.01 = 1010
-        let minProfitPercentage = ethers.utils.parseEther(Config.vars.minProfitPercentage);
         const decimals9 = BigNumber.from('1000000000');
         const decimal18 = ethers.utils.parseEther('1');
         const decimals27 = ethers.utils.parseEther('1000000000');
+        let minProfitPercentage = ethers.utils.parseEther(Config.vars.minProfitPercentage);
 
+        let calc = auction.price.mul(minProfitPercentage);
+        let priceWithProfit = calc.div(decimal18);
 
+
+        //adjusting lot to lotDaiValue
+        let lotDaiValue = ethers.utils.parseEther(Config.vars.lotDaiValue).mul(decimal18);
+        let minLot = lotDaiValue.div(auction.price.div(decimals9));
```

Comment:
Not sure I understand this logic.  Wouldn't `lotDaiValue` refer to a maximum lot size?  I'd assume we'd want to take the minimum of the auction's lot size and this user-imposed maximum lot size, and `fetch` that.